### PR TITLE
Improve shutdown of event reactor

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR removes the delay on shutdown caused by the `event::Reactor` using the combination of a running flag and reacting to that flag based on the, as well as signalling and joining shutdown in the same function (delaying other shutdown activities).
 

- replaces the use of the atomic boolean as a  stop/close/disconnect signal with a sender that takes a union type, WebSocketClientCmd.  This union type is either a Frame or the stop - signal.
 This change makes the shutdown essentially immediate, instead of waiting  for a message from the server (most-likely a ping) before disconnecting the websocket
- splits `event::Reactor` shutdown function into two parts: a shutdown signaler and a wait-for-shutdown function.  This allows the signaling to be done in a non-blocking model and the wait-for-shutdown to be done as part of a join process.

To test:

Build and up gameroom:

```
$ docker-compose -f examples/gameroom/docker-compose.yaml up --build
```

Once it has started and the admin websockets have been created, in another terminal:

```
$ docker-compose -f examples/gameroom/docker-compose.yaml stop gameroomd-acme
```
This should exit `0`, instead of `137`.

### CODE REVIEW

File: libsplinter/src/events/reactor.rs [hunk 1]
```diff
@@ -135,15 +136,29 @@ impl Reactor {
         }
     }
 
+    /// Return a ReactorShutdownSignaler, for use to send messages to
```
Comment (agunde406) [inline libsplinter/src/events/reactor.rs]: This comment is confusing. Can it be reworded?

Comment (peterschwarz) [reply libsplinter/src/events/reactor.rs]: Updated

File: examples/gameroom/daemon/src/main.rs [hunk 2]
```diff
@@ -122,9 +122,15 @@ fn run() -> Result<(), GameroomDaemonError> {
         public_key.as_hex(),
     )?;
 
+    let reactor_shutdown_signaler = reactor.shutdown_signaler();
+
     ctrlc::set_handler(move || {
         info!("Received Shutdown");
 
+        if let Err(err) = reactor_shutdown_signaler.signal_shutdown() {
+            error!("Unable to cleanly shutdown Event Reactor: {}", err);
```
Comment (jsmitchell) [inline examples/gameroom/daemon/src/main.rs]: Be consistent with capitalization of Event Reactor (reactor.rs logs it as "event reactor")

Comment (peterschwarz) [reply examples/gameroom/daemon/src/main.rs]: Updated.

**Pull request merged**